lib/integrations/calendar/services/CalDavCalendarService.ts

```diff
@@ -0,0 +1,10 @@
+import { Credential } from "@prisma/client";
+
+import { CALENDAR_INTEGRATIONS_TYPES } from "../constants/generals";
+import CalendarService from "./BaseCalendarService";
+
+export default class CalDavCalendarService extends CalendarService {
+  constructor(credential: Credential) {
```

Comment:
Just general observation that using inheritance to build a factory pattern is a way to do it I suppose, but it results in a lot of bloat.

Reply:
I find it very neat being able to reuse as much logic as possible. How is this bloated?